File: 151-ReverseWordsInAString/151-ReverseWordsInAString.java
// Last updated: 6/22/2025, 11:04:38 AM
class Solution {
    public String reverseWords(String s) {
        s=s.trim();
        StringBuilder res=new StringBuilder();
        for(int i=s.length()-1;i>=0;i--){
            if(s.charAt(i)==' ') continue;
            int j=i;
            while(j>=0 && s.charAt(j)!=' '){
                j--;
            }
            res.append(s.substring(j + 1, i + 1)).append(" ");
            i = j;
        }
        return res.toString().trim();
    }
}

File: 151-ReverseWordsInAString/151-ReverseWordsInAString.cpp
// Last updated: 6/22/2025, 11:04:43 AM
class Solution {
public:
    string reverseWords(string s) {
        string word, res;
        istringstream stream(s); 

        while (stream >> word) { 
            if (!res.empty()) {
                res = " " + res;
            }
            res = word + res;
        }

        return res; 
    }
};
